src/components/PupilComponents/QuizResults/QuizResults.tsx

```diff
@@ -0,0 +1,163 @@
+  const { questionResult, quizQuestion } = props;
+  return (
+    <OakFlex $flexDirection={"column"}>
+      <OakSpan $font={"body-3-bold"}>Correct answer:</OakSpan>
+      <OakFlex $flexDirection={"column"}>
+        {isArray(questionResult?.correctAnswer) &&
+          questionResult?.correctAnswer?.map((correctAnswer, index) => {
+            return (
+              <MathJaxWrap>
+                <OakQuizResultItem
+                  key={index}
+                  standardText={correctAnswer}
+                  boldPrefixText={pickBoldPretext(
+                    quizQuestion?.questionType,
+                    index,
+                    quizQuestion?.answers,
+                  )}
+                />
+              </MathJaxWrap>
+            );
+          })}
+      </OakFlex>
+    </OakFlex>
+  );
+};
+
+type ResultsInnerProps = {
+  index: number;
+  quizArray: QuestionsArray;
+  questionResult: QuestionState;
+  lessonSection: "exit-quiz" | "starter-quiz";
+};
+
+const ResultsInner = (props: ResultsInnerProps) => {
+  const { index, quizArray, questionResult, lessonSection } = props;
+  const quizQuestion = quizArray[index];
+  const questionStem = quizQuestion?.questionStem;
+  const answers = quizQuestion?.answers;
+  const grade = questionResult.grade;
+  return (
+    <OakFlex
+      key={`${lessonSection}-question-${index}`}
+      $gap={"space-between-s"}
+      $flexDirection={["column", "row"]}
+      role="listitem"
+    >
+      <OakIcon
+        iconName={grade === 1 ? "tick" : "cross"}
+        $background={grade === 1 ? "icon-success" : "icon-error"}
+        $colorFilter={"white"}
+        $borderRadius={"border-radius-circle"}
+      />
+      <OakFlex $flexDirection={"column"} $gap={"space-between-m"}>
+        {questionStem && (
+          <QuizResultQuestionStem index={index} questionStem={questionStem} />
+        )}
+        {answers?.["multiple-choice"] && isArray(questionResult?.feedback) && (
+          <QuizResultMCQ
+            answers={answers["multiple-choice"]}
+            feedback={questionResult.feedback}
+            pupilAnswer={questionResult.pupilAnswer}
+          />
+        )}
+        {answers?.["short-answer"] &&
+          !isArray(questionResult?.feedback) &&
+          questionResult?.feedback && (
+            <QuizResultShortAnswer
+              pupilAnswer={questionResult.pupilAnswer}
+              feedback={questionResult.feedback}
+            />
+          )}
+        {answers?.["order"] && isArray(questionResult?.feedback) && (
+          <QuizResultOrder
+            answers={answers["order"]}
+            feedback={questionResult.feedback}
+            pupilAnswers={questionResult.pupilAnswer}
+          />
+        )}
+        {answers?.["match"] && isArray(questionResult?.feedback) && (
+          <QuizResultMatch
+            answers={answers["match"]}
+            feedback={questionResult.feedback}
+            pupilAnswers={questionResult.pupilAnswer}
```

Comment:
this is still an array of strings at  this point. I'm not sure how TS is letting you get away with it but you basically have an implicit cast from string to int here. What you need to do  at this point in the code is parseInt and check for NaN results.

Reply:
i was planning on doing this but ts didnt complain, so i left it